Implement CodeInside Web project

anastasiiachervotochenko opened the pull request.

Commit:
Initial commit

Commit:
Add basic structure

anastasiia-zin commented on codeInside/src/index.html:
```diff
@@ -0,0 +1,13 @@
+  <title>CodeInside</title>
+  <base href="/">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <!-- <link rel="icon" type="image/x-icon" href="favicon.ico"> -->
```
maybe better to remove or uncomment, or leave with TODO? 
the same for other

anastasiiachervotochenko replied:
done

mgtippet33 committed:
add registration page

mgtippet33 committed:
added api for registration

mgtippet33 committed:
fixed registration api
added some validators

Commit:
Update .gitignore

Commit:
Add form validation and main page

mgtippet33 committed:
added todo

anastasiiachervotochenko committed:
Update index.html

Commit:
Fix login validation

mgtippet33 committed:
fixed some bugs

mgtippet33 committed:
Merge branch 'dev-sprint#1' into dev-register

mgtippet33 committed:
Merge pull request #2 from anastasiiachervotochenko/dev-register

Dev register

Commit:
Merge branch 'dev-sprint#1' of https://github.com/anastasiiachervotochenko/CodeInside-Web into dev-sprint#1

Commit:
remove white space

mgtippet33 committed:
added login api

thekorobov committed:
Added initial page

anastasiiachervotochenko committed:
Merge branch 'dev-sprint#1' into dev-sprint#0

anastasiiachervotochenko committed:
Merge pull request #3 from anastasiiachervotochenko/dev-sprint#0

Add main page

Commit:
Set up validation

Commit:
Merge branch 'dev-sprint#1' of https://github.com/anastasiiachervotochenko/CodeInside-Web into dev-sprint#1

Commit:
Update login.component.scss

Commit:
Update register.component.ts

mgtippet33 committed:
fixed some bugs from API in login and registration pages

anastasiiachervotochenko committed:
Merge pull request #4 from anastasiiachervotochenko/dev-sprint#1

Dev sprint#1

avzinovieva committed:
fixed font family

avzinovieva committed:
fixed styles on login and register

avzinovieva committed:
started implementing task solution page, added markup for discussion block

avzinovieva committed:
added angular and package-lock files

anastasiia-zin committed:
Merge pull request #5 from anastasiiachervotochenko/discusionBlock

Discusion block

mgtippet33 committed:
fix discussion block
add comment api

anastasiiachervotochenko committed:
Task list page (#6)

* Update register.component.ts

* Initial commit

* Implement structure of task list page

* add user timezone

* add api for tasks and topics

* Update mainPage.component.ts

* Rename page component, add range slider component and select dropdown

* Rename taskPage

* Fix bugs, add search field, refactor code

Co-authored-by: Matvii <72960397+mgtippet33@users.noreply.github.com>

anastasiiachervotochenko committed:
Implement Theory page

mgtippet33 committed:
Merge branch 'discussion-block' into dev-sprint#2

mgtippet33 committed:
Merge remote-tracking branch 'origin/topicPage' into dev-sprint#2

mgtippet33 committed:
fix bugs on image

mgtippet33 committed:
add dynamic navbar
rename topic on theory
add api for task, theory pages

mgtippet33 committed:
fixed register api
fixed login and register logic

mgtippet33 committed:
add abbreviation description for task view list

Commit:
Merge branch 'dev-sprint#2' of https://github.com/anastasiiachervotochenko/CodeInside-Web into dev-sprint#2

anastasiiachervotochenko committed:
Dev sprint#2 (#10)

* Update register.component.ts

* fixed font family

* fixed styles on login and register

* started implementing task solution page, added markup for discussion block

* added angular and package-lock files

* fix discussion block
add comment api

* Task list page (#6)

* Update register.component.ts

* Initial commit

* Implement structure of task list page

* add user timezone

* add api for tasks and topics

* Update mainPage.component.ts

* Rename page component, add range slider component and select dropdown

* Rename taskPage

* Fix bugs, add search field, refactor code

Co-authored-by: Matvii <72960397+mgtippet33@users.noreply.github.com>

* Implement Theory page

* Merge remote-tracking branch 'origin/topicPage' into dev-sprint#2

* fix bugs on image

* add dynamic navbar
rename topic on theory
add api for task, theory pages

* fixed register api
fixed login and register logic

* add abbreviation description for task view list

Co-authored-by: Anastasiia <avzinovyeva62@gmail.com>
Co-authored-by: anastasiia-zin <80216438+anastasiia-zin@users.noreply.github.com>
Co-authored-by: Matvii <72960397+mgtippet33@users.noreply.github.com>

mgtippet33 committed:
fixed comments in discussion
added api for getting user profile
added check for long-term user authorization

mgtippet33 committed:
fixed css style in welcome page

mgtippet33 committed:
implemented task solving on the task page
small fix on the theory page

mgtippet33 committed:
fix content flex on task page

mgtippet33 committed:
added submissions api
connected submissions api for task page

mgtippet33 committed:
added check for solving task

mgtippet33 committed:
turn off authorization checker

thekorobov committed:
Update search

thekorobov committed:
Update taskViewPage.component.scss

thekorobov committed:
Update theoryViewPage.component.scss

thekorobov committed:
Update theoryPage.component.scss

thekorobov committed:
Updated text print

thekorobov committed:
Update theoryViewPage.component.ts

mgtippet33 committed:
Merge pull request #12 from anastasiiachervotochenko/taskSolutionPage

Task solution page

thekorobov committed:
Update theoryViewPage.component.ts

Commit:
Merge branch 'dev-sprint#3' of https://github.com/anastasiiachervotochenko/CodeInside-Web into dev-sprint#3

Commit:
Add soring for task list page

Commit:
Merge branch 'dev-sprint#3' of https://github.com/anastasiiachervotochenko/CodeInside-Web into dev-sprint#3

mgtippet33 committed:
Merge pull request #13 from mgtippet33/dev-sprint#3

Dev sprint#3

mgtippet33 committed:
added CI action for build testing
prepared the project for deploy
restructured project files

mgtippet33 committed:
fixed auto-deployment

mgtippet33 committed:
install express framework

mgtippet33 committed:
fixed some issues

mgtippet33 committed:
changed ApiConstants and http.service

mgtippet33 committed:
fixed CORS Privacy problem

mgtippet33 committed:
install cors dependencies

mgtippet33 committed:
fixed CORS problem#1

mgtippet33 committed:
fixed CORS problem#2

mgtippet33 committed:
fixed CORS problem#3

mgtippet33 committed:
fixed CORS problem#4

mgtippet33 committed:
restructured project files

avzinovieva committed:
added profile page

avzinovieva committed:
added achievement page

mgtippet33 committed:
added modal view on task page

mgtippet33 committed:
restructured project files

mgtippet33 committed:
Merge branch 'dev-sprint#4' into userProfile

mgtippet33 committed:
changed api format

mgtippet33 committed:
added api for payment
added api for edited profile

mgtippet33 committed:
connected edited for task

mgtippet33 committed:
added update/remove tasks
added hint modal
added remove user comment

mgtippet33 committed:
add create task modal
fixed some bugs in task page

mgtippet33 committed:
added CRUD for theory

mgtippet33 committed:
Merge pull request #16 from mgtippet33/deploy-workflow

Prepare to deploy

Commit:
Fix date displaying

mgtippet33 committed:
added redirect to profile via comment
added comment view for admins without task solution

thekorobov committed:
Fixed achievements

anastasiia-zin committed:
Merge pull request #17 from mgtippet33/crud-task-page

CRUD for tasks and theories

avzinovieva committed:
added ban user and set moderator buttons

mgtippet33 committed:
Merge branch 'dev-sprint#4' of into achievementPage

Commit:
Add other users pages

mgtippet33 committed:
Merge pull request #15 from mgtippet33/achievementPage

Achievement page

mgtippet33 committed:
fixed some bugs

mgtippet33 committed:
Merge branch 'dev-sprint#4' of  into userProfile

mgtippet33 committed:
Merge pull request #14 from mgtippet33/userProfile

added profile page

mgtippet33 committed:
fixed bugs

mgtippet33 committed:
fixed task css

mgtippet33 committed:
added redesign

mgtippet33 committed:
Merge branch 'dev-sprint#4' of into web-redesign

mgtippet33 committed:
added icons for crud

mgtippet33 committed:
added notification modal for login and profile pages
added modifiable button to ban / unban users
fixed log out

mgtippet33 committed:
added reset password

mgtippet33 committed:
code cleanup

mgtippet33 committed:
fixed set cookie

mgtippet33 committed:
Merge pull request #19 from mgtippet33/web-redesign

Web redesign

mgtippet33 committed:
changed main url

mgtippet33 committed:
Merge branch 'dev-sprint#4' of into dev

mgtippet33 committed:
changed favicon

mgtippet33 committed:
increased configuration budgets

mgtippet33 committed:
fixed project build

mgtippet33 committed:
fixed project build #2

mgtippet33 committed:
fixed project build #3

mgtippet33 committed:
added jquery and popperjs

mgtippet33 committed:
added new design

mgtippet33 committed:
added photo change function
added adaptation of the photo in the profile and in the comments

mgtippet33 committed:
added check for adding photos

mgtippet33 committed:
Merge branch 'master' into dev
